x/ibc/02-client/client/cli/tx.go

```diff
@@ -0,0 +1,155 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+)
+
+// GetTxCmd returns the transaction commands for IBC Clients
+func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	ics02ClientTxCmd := &cobra.Command{
+		Use:                        "client",
+		Short:                      "Client transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+	}
+
+	ics02ClientTxCmd.AddCommand(client.PostCommands(
+		GetCmdCreateClient(cdc),
+		GetCmdUpdateClient(cdc),
+	)...)
+
+	return ics02ClientTxCmd
+}
+
+// GetCmdCreateClient defines the command to create a new IBC Client as defined
+// in https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#create
+func GetCmdCreateClient(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [client-id] [path/to/consensus_state.json]",
+		Short: "create new client with a consensus state",
+		Long: strings.TrimSpace(fmt.Sprintf(`create new client with a specified identifier and consensus state:
+
+Example:
+$ %s tx ibc client create [client-id] [path/to/consensus_state.json] --from node0 --home ../node0/<app>cli --chain-id $CID
+		`, version.ClientName),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			contents, err := ioutil.ReadFile(args[1])
+			if err != nil {
+				return err
+			}
+
+			var state exported.ConsensusState
+			if err := cdc.UnmarshalJSON(contents, &state); err != nil {
+				return err
+			}
+
+			msg := types.MsgCreateClient{
+				ClientID: args[0],
+
+				ConsensusState: state,
+				Signer:         cliCtx.GetFromAddress(),
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
+
+// GetCmdUpdateClient defines the command to update a client as defined in
+// https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#update
+func GetCmdUpdateClient(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update [client-id] [path/to/header.json]",
+		Short: "update existing client with a header",
+		Long: strings.TrimSpace(fmt.Sprintf(`update existing client with a header:
+
+Example:
+$ %s tx ibc client create [client-id] [path/to/header.json] --from node0 --home ../node0/<app>cli --chain-id $CID
+		`, version.ClientName),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			bz, err := ioutil.ReadFile(args[1])
```

Comment:
This command should take either:
- The `json` emitted by `gaiacli q ibc client header`
- A file path to the data emitted by `gaiacli q ibc client header`

Maybe something like:

```go
			var header client.Header
			if err := cdc.UnmarshalJSON([]byte(args[1]), &header); err != nil {
				fmt.Fprintf(os.Stderr, "failed to unmarshall input into struct, checking for file...")
				contents, err := ioutil.ReadFile(args[1])
				if err != nil {
					return fmt.Errorf("error opening header file: %v\n", err)
				}
				if err := cdc.UnmarshalJSON(contents, &header); err != nil {
					return fmt.Errorf("error unmarshalling header file: %v\n", err)
				}
			}
```